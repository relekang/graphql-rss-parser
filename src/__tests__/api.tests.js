const { testGraphqlApi } = require('./utils')

const websiteUrl = 'https://rolflekang.com'
const url = 'https://rolflekang.com/feed.xml'

testGraphqlApi`
  feed(url: "${url}") { title }
`

testGraphqlApi`
  feed(url: "${url}") {
    title
    link
    entries {
      title
      link
      pubDate
    }
  }
`

testGraphqlApi`
  feed(url: "https://rolflekang.com/not-a-feed.xml") { title }
`

testGraphqlApi`
  feed(url: "https://non-existing-domain.com") { title }
`
testGraphqlApi`
  feed(url: "https://non-existing-domain.com") { title, badField }
`

testGraphqlApi`
  fromWebsite(websiteUrl: "${websiteUrl}") {
    link
  }
`
