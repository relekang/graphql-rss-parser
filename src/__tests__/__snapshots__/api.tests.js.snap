// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`
  feed(url: "https://non-existing-domain.com") { title }
 1`] = `
Object {
  "data": Object {
    "feed": null,
  },
  "errors": Array [
    Object {
      "message": "getaddrinfo ENOTFOUND non-existing-domain.com non-existing-domain.com:443",
      "stack": null,
    },
  ],
}
`;

exports[`
  feed(url: "https://non-existing-domain.com") { title, badField }
 1`] = `
Object {
  "errors": Array [
    Object {
      "message": "Cannot query field \\"badField\\" on type \\"Feed\\".",
      "stack": null,
    },
  ],
}
`;

exports[`
  feed(url: "https://rolflekang.com/feed.xml") {
    title
    link
    entries {
      title
      link
      pubDate
    }
  }
 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "entries": Array [
        Object {
          "link": "http://rolflekang.com/lets-run-some-races",
          "pubDate": "2017-01-02T00:00:00.000Z",
          "title": "Let's run some races this year",
        },
        Object {
          "link": "http://rolflekang.com/software-development-on-an-ipad",
          "pubDate": "2017-01-01T00:00:00.000Z",
          "title": "Software development on an iPad",
        },
        Object {
          "link": "http://rolflekang.com/filtering-lint-errors",
          "pubDate": "2016-06-10T00:00:00.000Z",
          "title": "Filtering lint errors",
        },
        Object {
          "link": "http://rolflekang.com/interrail-summer",
          "pubDate": "2015-07-26T00:00:00.000Z",
          "title": "Interrail summer",
        },
        Object {
          "link": "http://rolflekang.com/writing-latex-in-atom",
          "pubDate": "2015-05-23T00:00:00.000Z",
          "title": "Writing (latex) in Atom",
        },
        Object {
          "link": "http://rolflekang.com/rmoq",
          "pubDate": "2014-12-30T00:00:00.000Z",
          "title": "rmoq - a request mock cache",
        },
        Object {
          "link": "http://rolflekang.com/django-nopassword-one-point-o",
          "pubDate": "2014-10-03T15:35:00.000Z",
          "title": "django-nopassword reached 1.0",
        },
        Object {
          "link": "http://rolflekang.com/moving-to-pelican",
          "pubDate": "2014-05-07T00:00:00.000Z",
          "title": "Moving the blog to pelican",
        },
        Object {
          "link": "http://rolflekang.com/building-tumblr-theme-with-grunt",
          "pubDate": "2014-02-02T00:00:00.000Z",
          "title": "Building a Tumblr theme with GruntJS",
        },
        Object {
          "link": "http://rolflekang.com/backup-routines-for-redisdb",
          "pubDate": "2013-10-24T00:00:00.000Z",
          "title": "Backup routine for Redis",
        },
        Object {
          "link": "http://rolflekang.com/django-app-custom-user",
          "pubDate": "2013-05-22T00:00:00.000Z",
          "title": "Making your Django app ready for custom users",
        },
        Object {
          "link": "http://rolflekang.com/stats-are-fun",
          "pubDate": "2013-03-03T00:00:00.000Z",
          "title": "Stats are fun",
        },
        Object {
          "link": "http://rolflekang.com/alfred-scripts-for-jekyll",
          "pubDate": "2013-02-26T00:00:00.000Z",
          "title": "Alfred scripts for Jekyll",
        },
        Object {
          "link": "http://rolflekang.com/use-array-in-postgresql-with-django",
          "pubDate": "2013-01-27T00:00:00.000Z",
          "title": "Use array in postgresql with django",
        },
        Object {
          "link": "http://rolflekang.com/apply-puppet-configuration-automatically",
          "pubDate": "2013-01-03T00:00:00.000Z",
          "title": "Apply puppet automatically",
        },
        Object {
          "link": "http://rolflekang.com/reading-wikipedia",
          "pubDate": "2012-12-02T00:00:00.000Z",
          "title": "Readable Wikipedia",
        },
        Object {
          "link": "http://rolflekang.com/django-nopassword",
          "pubDate": "2012-09-28T00:00:00.000Z",
          "title": "django-nopassword",
        },
        Object {
          "link": "http://rolflekang.com/the-github-lamp",
          "pubDate": "2012-09-16T00:00:00.000Z",
          "title": "The Github lamp",
        },
      ],
      "link": "http://rolflekang.com/",
      "title": "Writings of Rolf Erik Lekang",
    },
  },
}
`;

exports[`
  feed(url: "https://rolflekang.com/feed.xml") { title }
 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "title": "Writings of Rolf Erik Lekang",
    },
  },
}
`;

exports[`
  feed(url: "https://rolflekang.com/not-a-feed.xml") { title }
 1`] = `
Object {
  "data": Object {
    "feed": null,
  },
  "errors": Array [
    Object {
      "message": "Not Found",
      "stack": null,
    },
  ],
}
`;

exports[`
  fromWebsite(websiteUrl: "https://rolflekang.com") {
    link
  }
 1`] = `
Object {
  "data": Object {
    "fromWebsite": Array [
      Object {
        "link": "https://rolflekang.com/feed.xml",
      },
    ],
  },
}
`;
